example/particle_leapfrog/particle_leapfrog.cpp

```diff
@@ -158,23 +158,54 @@ TaskStatus WriteParticleLog(BlockList_t &blocks, int ncycle) {
     const auto &x = swarm->Get<Real>("x").Get().GetHostMirrorAndCopy();
     const auto &y = swarm->Get<Real>("y").Get().GetHostMirrorAndCopy();
     const auto &z = swarm->Get<Real>("z").Get().GetHostMirrorAndCopy();
-    const auto &vx = swarm->Get<Real>("vx").Get().GetHostMirrorAndCopy();
-    const auto &vy = swarm->Get<Real>("vy").Get().GetHostMirrorAndCopy();
-    const auto &vz = swarm->Get<Real>("vz").Get().GetHostMirrorAndCopy();
-
-    const auto &is_active = swarm->GetMask().Get().GetHostMirrorAndCopy();
-    for (auto n = 0; n < is_active.GetSize(); n++) {
+    const auto &v = swarm->Get<Real>("v").Get().GetHostMirrorAndCopy();
+    const auto &vv = swarm->Get<Real>("vv").Get().GetHostMirrorAndCopy();
+
+    PackIndexMap imap;
+    std::vector<std::string> pack_names = {"v", "vv"};
+    auto vp = swarm->PackVariables<Real>(pack_names, imap);
+    auto iv = imap.GetFlatIdx("v");
+    auto ivv = imap.GetFlatIdx("vv");
+
+    const auto &is_active =
+        Kokkos::create_mirror_view_and_copy(HostMemSpace(), swarm->GetMask());
+    for (auto n = 0; n <= swarm->GetMaxActiveIndex(); n++) {
       if (is_active(n)) {
         particle_output_this_rank(offset++) = static_cast<Real>(pmb->gid);
         particle_output_this_rank(offset++) = static_cast<Real>(id(n));
         particle_output_this_rank(offset++) = x(n);
         particle_output_this_rank(offset++) = y(n);
         particle_output_this_rank(offset++) = z(n);
-        particle_output_this_rank(offset++) = vx(n);
-        particle_output_this_rank(offset++) = vy(n);
-        particle_output_this_rank(offset++) = vz(n);
+        particle_output_this_rank(offset++) = v(0, n);
+        particle_output_this_rank(offset++) = v(1, n);
+        particle_output_this_rank(offset++) = v(2, n);
+
+        // Check that vv is still consistent with v
+        for (int i = 0; i < 3; i++) {
+          for (int j = 0; j < 3; j++) {
+            if (std::fabs(vv(i, j, n)) > 1.e-10) {
```

Comment:
Why the check for `>1e-10`?

Reply:
Oops, this is a leftover from using a soft equivalence (which can get upset when one of the arguments is exactly `0`. Removed.